src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Instrument.daml

```diff
@@ -89,12 +97,19 @@ template Instrument
             let l = last calculationSchedule
             assertMsg "The adjusted terminationDate must match the adjusted end date of the last period" $ l.adjustedEndDate == terminationDateAdjusted
 
+            -- If there is an fxLinkedNotionalSchedule, look up the base notional from a different swap leg
+            let
```

Comment:
```suggestion
            let constNotionalScheduleReference = getNotionalFromOtherSwapStream . (.constantNotionalScheduleReference) <$> c.fxLinkedNotionalSchedule
```

Reply:
I tried this, but seems it causes an error on row 108